lib/side_menu_animation.dart

```diff
@@ -196,43 +204,85 @@ class _SideMenuAnimationState extends State<SideMenuAnimation>
-                              if (i != 0) {
-                                setState(() {
-                                  _oldSelectedIndex = _selectedIndex;
-                                  _selectedIndex = i - 1;
-                                  _selectedColor = i;
-                                });
-                                _dontAnimate = false;
-                              } else {
-                                _dontAnimate = true;
-                              }
-                              widget.onItemSelected(i);
-                            },
-                            child: widget.items[i],
-                          ),
-                        ),
-                      ),
-                    ),
+                    // ignore: unrelated_type_equality_checks
+
+                    (widget.position != Position.left)
+                        ? Positioned(
+                            right: 0,
+                            top: itemSize * i,
+                            width: widget.menuWidth,
+                            height: itemSize,
+                            child: Transform(
+                              transform: Matrix4.identity()
+                                ..setEntry(3, 2, 0.001)
+                                ..rotateY(_animationController.status ==
+                                        AnimationStatus.reverse
+                                    ? -_animations[widget.items.length - 1 - i]
+                                        .value
+                                    : -_animations[i].value),
+                              alignment: Alignment.topRight,
+                              child: Material(
+                                color: (i == _selectedColor)
+                                    ? widget.selectedColor
+                                    : widget.unselectedColor,
+                                child: InkWell(
+                                  onTap: () {
+                                    _animationController.forward(from: 0.0);
+                                    if (i != 0) {
+                                      setState(() {
+                                        _oldSelectedIndex = _selectedIndex;
+                                        _selectedIndex = i - 1;
+                                        _selectedColor = i;
+                                      });
+                                      _dontAnimate = false;
+                                    } else {
+                                      _dontAnimate = true;
+                                    }
+                                    widget.onItemSelected(i);
+                                  },
+                                  child: widget.items[i],
+                                ),
+                              ),
+                            ),
+                          )
+                        : Positioned(
```

Comment:
There are a lot of code that could be reused here, please avoid creating the same code twice.